plugin/manifest_test.go

```diff
@@ -82,6 +80,101 @@ func TestManifest_ResolvePath(t *testing.T) {
 	}
 }
 
+func TestManifest_ResolveOptions(t *testing.T) {
+
+	defaultValue := "default"
+	tests := []struct {
+		name         string
+		testManifest Manifest
+		selections   map[string]string
+		wantError    string
+		wantOptions  map[string]string
+	}{
+		{
+			name: "Success/AllDefaults",
+			testManifest: Manifest{
+				ExecutablePath: "testplugin",
+				Configuration: []ConfigurationOption{
+					{
+						Name:        "default",
+						Description: "A required options",
+						Required:    false,
+						Default:     &defaultValue,
+					},
+				},
+			},
+			wantOptions: map[string]string{"default": "default"},
+		},
+		{
+			name: "Success/WithSelections",
+			testManifest: Manifest{
+				ExecutablePath: "testplugin",
+				Configuration: []ConfigurationOption{
+					{
+						Name:        "required",
+						Description: "A required options",
+						Required:    true,
+					},
+					{
+						Name:        "default",
+						Description: "A default option",
+						Required:    false,
+						Default:     &defaultValue,
+					},
+					{
+						Name:        "default2",
+						Description: "A default option",
+						Required:    false,
+						Default:     &defaultValue,
+					},
```

Comment:
Maybe you could also include a case with a required option and explicit default value.